pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1484,38 +1484,11 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 			var pciAddr string
 			pciAddr, sriovPciAddresses, err = popSRIOVPCIAddress(iface.Name, sriovPciAddresses)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to configure SRIOV %s: %v", iface.Name, err)
```

Comment:
Seems the pattern you are trying to advocate here is for the error at the caller to describe what operation failed by describing the called function. The patter which I think fits better is for the function/object to describe itself (it knows best) and the caller is left to just expressing what failed at its abstraction level and adding relevant info which is not provided by the error itself.

In this case, we add the iface name and express that the SRIOV configuration failed.
The details on why it failed is in the error itself.
This seems to work fine here.

Reply:
Yep this what I meant, but I dont expect the a function to describe the called function or what it do.
I want a function to describe what failed at its abstraction level like you said.
This function is very long and it is hard to follow, failed to configure SRIOV is not enough info especially when you have the exect same error message more then one time (line 1492) it make debugging this very hard giving a bit more info will make it much better.

In popSRIOVPCIAddress function case, we can compromise since it returns only one error that give enough information, but then again if there will be a change for popSRIOVPCIAddress that will make it generate another error it will not reflect in the logs what exactly went wrong

Reply:
I do not understand your point here, sorry. This error is returned from `Convert_v1_VirtualMachine_To_api_Domain`, so at its abstraction level it can say "failed convert from vmi to domain" or just follow the existing format of "failed configure SRIOV" as it was before. This portion of the code indeed handles SRIOV and the info added follows the existing usages here.

The suggested text you gave originally tries to express what the called function does, not what its failure caused.
So I am not clear what exactly is the AI on this comment now.

Reply:
My point is `Convert_v1_VirtualMachine_To_api_Domain` return the exact same error in two different places,
in case of an error it will be hard to understand where the code failed

Reply:
How will it return the same exact error? The error returned by `popSRIOVPCIAddress` includes unique information about what actually failed and that is chained to the declaration that the SRIOV failed.